src/protocols/secure_channel/CASESession.cpp

```diff
@@ -562,40 +623,43 @@ CHIP_ERROR CASESession::SendSigmaR2()
     return err;
 }
 
-CHIP_ERROR CASESession::HandleSigmaR2_and_SendSigmaR3(const System::PacketBufferHandle & msg)
+CHIP_ERROR CASESession::HandleSigmaR2_and_SendSigmaR3(System::PacketBufferHandle & msg)
 {
     ReturnErrorOnFailure(HandleSigmaR2(msg));
     ReturnErrorOnFailure(SendSigmaR3());
 
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR CASESession::HandleSigmaR2(const System::PacketBufferHandle & msg)
+CHIP_ERROR CASESession::HandleSigmaR2(System::PacketBufferHandle & msg)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
+    System::PacketBufferTLVReader tlvReader;
+    System::PacketBufferTLVReader suppTlvReader;
+    TLV::TLVType containerType = TLV::kTLVType_Structure;
 
     const uint8_t * buf = msg->Start();
     size_t buflen       = msg->DataLength();
 
     chip::Platform::ScopedMemoryBuffer<uint8_t> msg_salt;
     uint16_t saltlen;
 
-    chip::Platform::ScopedMemoryBuffer<uint8_t> msg_R2_Signed;
+    System::PacketBufferHandle msg_R2_Encrypted;
```

Comment:
Let's not use PacketBufferHandle for internal buffers that are not actually sent. ScopedMemoryBuffers are better suited for such allocations.

The system could have limited PacketBufferHandle pool and we should use it only when necessary.

This comment applies to multiple areas in this file.

Reply:
Replaced internal buffers using PacketBufferHandle with ScopedMemoryBuffers.